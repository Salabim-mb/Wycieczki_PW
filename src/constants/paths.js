export const paths = {
	DASHBOARD: {
		name: "Strona główna",
		path: "/",
	},
	BLOG: {
		name: 'Blog',
		path: '/:blog',
	},
	BLOG_CATEGORIES: {
		name: "Kategorie",
		path: "/blog/:category",
		redirect: (newCategory) => `/blog/${newCategory}`,
	},
	BLOG_ENTRY: {
		name: "Wpis",
		path: "/blog/:category/:entry_id",
		redirect: (category, entryId) => `/blog/${category}/${entryId}`,
	},
	POST_CREATOR: {
		name: 'Kreator wpisów',
		path: '/blog/post-creator/:id?',
		redirect: (entryId) => `/blog/post-creator/${entryId}`
	},
	CREATOR_CATEGORY: {
		name: "Krator kategorii",
		path: "/blog/category-creator/:id?",
		redirect: (categoryId) => `/blog/category-creator/${categoryId}`
	}
}