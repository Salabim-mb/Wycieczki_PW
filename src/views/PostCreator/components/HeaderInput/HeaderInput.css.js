import styled from 'styled-components';
import { FormGroup, Typography } from '@material-ui/core';
import { Input } from 'components';

export const StyledFormGroup = styled(FormGroup)`
	display: flex;
	flex-direction: column;
	margin-top: 1rem;
	@media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
		width: 100%;
		flex-direction: row;
		align-items: center;
	}
`;

export const StyledTypography = styled(Typography)`
	margin: 0.5rem 0;
	@media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
		margin: 0 1rem;
	}
`;

export const StyledInput = styled(Input)`
	width: 100%;
	@media (min-width: ${({ theme }) => theme.breakpoints.tablet}) {
		width: 350px;
	}
`;
