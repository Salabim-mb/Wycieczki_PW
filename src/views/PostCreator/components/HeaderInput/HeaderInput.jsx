import React from 'react';
import { Button, FormGroup, TextField, Typography } from '@material-ui/core';
import { PhotoCamera } from '@material-ui/icons';
import { Alert } from '@material-ui/lab';

const HeaderInput = ({ handleChangeFile, header, handleChangeLink, error }) => {
	const FILE_LABEL = '';
	return (
		<FormGroup my={2}>
			<Button variant="contained" component="label" color="default" startIcon={<PhotoCamera />}>
				Prześlij nagłówek
				<input
					id="headerFile"
					name="headerFile"
					type="file"
					accept="image/*"
					hidden
					onChange={handleChangeFile}
				/>
			</Button>
			<Typography my={2}>lub</Typography>
			<TextField
				fullWidth
				id="headerLink"
				name="headerLink"
				label="Link do filmu yt"
				variant="outlined"
				value={typeof header === 'string' ? header : FILE_LABEL}
				inputProps={{ minLength: 1, maxLength: 120 }}
				onChange={handleChangeLink}
			/>
			{error && <Alert severity="error">{error}</Alert>}
		</FormGroup>
	);
};

export default HeaderInput;
