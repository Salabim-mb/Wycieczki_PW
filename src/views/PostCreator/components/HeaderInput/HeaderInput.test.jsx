import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import HeaderInput from './HeaderInput';
import paths from 'constants/api';

describe('HeaderInput', () => {
	let props;

	beforeEach(() => {
		props = {
			handleChangeFile: jest.fn(),
			header: 'Nagłówek',
			handleChangeLink: jest.fn(),
			error: '',
		};
	});

	it('should match snapshot', () => {
		const { container } = render(<HeaderInput {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should display error message ', () => {
		props.error = 'błąd';
		const { container } = render(<HeaderInput {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should header value be empty', () => {
		props.header = paths.PLAIN;
		const { getByTestId } = render(<HeaderInput {...props} />);

		expect(getByTestId('test-id').value).toBe('');
	});
});
