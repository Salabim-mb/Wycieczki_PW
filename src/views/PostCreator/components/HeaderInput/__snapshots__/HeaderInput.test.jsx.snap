// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderInput should display error message  1`] = `
<div>
  <div
    class="MuiFormGroup-root"
    my="2"
  >
    <label
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="3.2"
            />
            <path
              d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
            />
          </svg>
        </span>
        Prześlij nagłówek
        <input
          accept="image/*"
          hidden=""
          id="headerFile"
          name="headerFile"
          type="file"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </label>
    <p
      class="MuiTypography-root MuiTypography-body1"
      my="2"
    >
      lub
    </p>
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink="true"
        for="headerLink"
        id="headerLink-label"
      >
        Link do filmu yt
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          data-testid="test-id"
          id="headerLink"
          maxlength="120"
          minlength="1"
          name="headerLink"
          type="text"
          value="Nagłówek"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
          >
            <span>
              Link do filmu yt
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        błąd
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderInput should match snapshot 1`] = `
<div>
  <div
    class="MuiFormGroup-root"
    my="2"
  >
    <label
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="3.2"
            />
            <path
              d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
            />
          </svg>
        </span>
        Prześlij nagłówek
        <input
          accept="image/*"
          hidden=""
          id="headerFile"
          name="headerFile"
          type="file"
        />
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </label>
    <p
      class="MuiTypography-root MuiTypography-body1"
      my="2"
    >
      lub
    </p>
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
        data-shrink="true"
        for="headerLink"
        id="headerLink-label"
      >
        Link do filmu yt
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          data-testid="test-id"
          id="headerLink"
          maxlength="120"
          minlength="1"
          name="headerLink"
          type="text"
          value="Nagłówek"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
          >
            <span>
              Link do filmu yt
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    
  </div>
</div>
`;
